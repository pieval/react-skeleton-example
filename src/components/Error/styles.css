.errorBox {
  border: 2px solid red;
  color: red;
  padding: 1em;
}

.closeButton {
  float: right;
}
