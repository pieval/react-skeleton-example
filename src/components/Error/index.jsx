import React, { PropTypes } from 'react';
import styles from './styles.css';

const Error = (props) => {
  const { message, onClose } = props;
  if (message) {
    return (
      <div className={styles.errorBox}>
        { message }
        <button className={styles.closeButton} onClick={onClose}>X</button>
      </div>
    );
  }
  return false;
};

Error.propTypes = {
  message: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default Error;
