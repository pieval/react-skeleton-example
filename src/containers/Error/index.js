import { connect } from 'react-redux';
import { resetError } from '../../redux/modules/error';
import Error from '../../components/Error';

export default connect(
  state => ({
    message: state.error.message,
  }),
  dispatch => ({
    onClose: () => dispatch(resetError()),
  }),
)(Error);
