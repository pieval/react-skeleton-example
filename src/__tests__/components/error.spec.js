/* eslint no-undef: 0 */
import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Error from '../../components/Error';

describe('<Error />', () => {
  describe('Rendering', () => {
    it('should render with default props / SNAPSHOT TEST', () => {
      const error = shallow(<Error message="Test error message" onClose={() => {}} />);
      expect(shallowToJson(error)).toMatchSnapshot();
    });
    it('should not render if message is empty', () => {
      const error = shallow(<Error message="" onClose={() => {}} />);
      expect(error.find('div .errorBox').exists()).toBeFalsy();
    });
  });
  describe('Close behavior', () => {
    let error;
    let onClose;

    beforeEach(() => {
      onClose = jest.fn();
      error = mount(<Error message="Test error message" onClose={onClose} />);
    });
    it('should has a onClose prop', () => {
      expect(error.props().onClose).toBeDefined();
      expect(error.props().onClose).toEqual(onClose);
    });
    it('should render a close button', () => {
      const button = error.find('button .closeButton');
      expect(button).toBeDefined();
    });
    it('should call onClose on close button click', () => {
      const button = error.find('button .closeButton');
      button.simulate('click');
      expect(onClose).toHaveBeenCalledTimes(1);
    });
  });
});
