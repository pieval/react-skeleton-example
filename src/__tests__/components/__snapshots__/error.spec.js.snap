// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> Rendering should render with default props / SNAPSHOT TEST 1`] = `
<div
  className="errorBox"
>
  Test error message
  <button
    className="closeButton"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
